src/tests/account/starknet/test_dual_account.cairo

```diff
@@ -85,35 +93,36 @@ fn test_dual_get_public_key() {
 }
 
 #[test]
+#[ignore] // REASON: inconsistent ENTRYPOINT_NOT_FOUND panic message
 #[should_panic(expected: ('ENTRYPOINT_NOT_FOUND',))]
 fn test_dual_no_get_public_key() {
     let dispatcher = setup_non_account();
     dispatcher.get_public_key();
 }
 
 #[test]
-#[should_panic(expected: ("Some error", 'ENTRYPOINT_FAILED',))]
+#[should_panic(expected: ("Some error",))]
 fn test_dual_get_public_key_exists_and_panics() {
-    let (dispatcher, _) = setup_account_panic();
-    dispatcher.get_public_key();
+    let (snake_dispatcher, _) = setup_account_panic();
+    snake_dispatcher.get_public_key();
 }
 
 #[test]
 fn test_dual_is_valid_signature() {
     let (snake_dispatcher, target) = setup_snake();
+    let data = SIGNED_TX_DATA(KEY_PAIR_2);
+    let signature = get_accept_ownership_signature(target.contract_address, PUBKEY, KEY_PAIR_2);
```

Comment:
Same answer :) Will update this test case accordingly once we make a decision

Reply:
I tried to keep the logic of the test case unchanged:

1. setup
2. set NEW_PUB_KEY as account's public key
3. check signature signed with NEW_PUB_KEY

For this flow we can't use `[data.r, data.s]` signature to set new public_key, because we need the special accept_ownership signature for that.

Now, looking into the logic it doesn't make much sense why we have to set a new public key prior to validating the signature. Perhaps, there was some idea behind it at the time of writing the test

We can either keep the logic as it is or get rid of the step of setting new public key and validate a signature signed with the key the account was deployed with. I vote for the latter, if there's no hidden sense in changing the public key

Reply:
The idea of setting the pub key was to reuse the signature we had to manually add at the time. Now we can sign in the test, but notice that in the eth account version of this, we are still setting the pub key, because we can't sign with the one set by default, since we don't have the keypair. We can refactor the code to deploy with the key pair by default, but I don't think we need to do that to get this PR merged.

Also, at least for consistency, we can get rid of the `let data = SIGNED_TX_DATA(KEY_PAIR_2);` line, and we can use  

```let serialized_signature = key_pair.serialized_sign(TRANSACTION_HASH.into());```

to get the signature, like we are doing in the eth account version of this test.

Reply:
Thx for explanation. I've updated the test case accordingly